Remove CSSTypedOMEnabled preference

annevk opened the pull request:
#### e225c278f4c06f451ea92cc68b12986dd2a99979
<pre>
Remove CSSTypedOMEnabled preference
<a href="https://bugs.webkit.org/show_bug.cgi?id=271128">https://bugs.webkit.org/show_bug.cgi?id=271128</a>

Reviewed by Tim Nguyen.

It&apos;s been stable for over a year.

* LayoutTests/css-typedom/attributeStyleMap.html:
* LayoutTests/css-typedom/css-style-value-parse.html:
* LayoutTests/css-typedom/sameobject.html:
* LayoutTests/fast/css-custom-paint/properties.html:
* LayoutTests/fast/css/css-typed-om/style-property-map-set-CSSMathSum-value.html:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/css/CSSStyleRule.idl:
* Source/WebCore/css/DOMCSSNamespace+CSSNumericFactory.idl:
* Source/WebCore/css/ElementCSSInlineStyle.idl:
* Source/WebCore/css/typedom/CSSKeywordValue.idl:
* Source/WebCore/css/typedom/CSSNumericValue.idl:
* Source/WebCore/css/typedom/CSSOMVariableReferenceValue.idl:
* Source/WebCore/css/typedom/CSSStyleImageValue.idl:
* Source/WebCore/css/typedom/CSSStyleValue.idl:
* Source/WebCore/css/typedom/CSSUnitValue.idl:
* Source/WebCore/css/typedom/CSSUnparsedValue.idl:
* Source/WebCore/css/typedom/StylePropertyMap.idl:
* Source/WebCore/css/typedom/StylePropertyMapReadOnly.idl:
* Source/WebCore/css/typedom/color/CSSColor.idl:
* Source/WebCore/css/typedom/color/CSSColorValue.idl:
* Source/WebCore/css/typedom/color/CSSHSL.idl:
* Source/WebCore/css/typedom/color/CSSHWB.idl:
* Source/WebCore/css/typedom/color/CSSLCH.idl:
* Source/WebCore/css/typedom/color/CSSLab.idl:
* Source/WebCore/css/typedom/color/CSSOKLCH.idl:
* Source/WebCore/css/typedom/color/CSSOKLab.idl:
* Source/WebCore/css/typedom/color/CSSRGB.idl:
* Source/WebCore/css/typedom/numeric/CSSMathClamp.idl:
* Source/WebCore/css/typedom/numeric/CSSMathInvert.idl:
* Source/WebCore/css/typedom/numeric/CSSMathMax.idl:
* Source/WebCore/css/typedom/numeric/CSSMathMin.idl:
* Source/WebCore/css/typedom/numeric/CSSMathNegate.idl:
* Source/WebCore/css/typedom/numeric/CSSMathProduct.idl:
* Source/WebCore/css/typedom/numeric/CSSMathSum.idl:
* Source/WebCore/css/typedom/numeric/CSSMathValue.idl:
* Source/WebCore/css/typedom/numeric/CSSNumericArray.idl:
* Source/WebCore/css/typedom/numeric/CSSNumericBaseType.idl:
* Source/WebCore/css/typedom/transform/CSSMatrixComponent.idl:
* Source/WebCore/css/typedom/transform/CSSPerspective.idl:
* Source/WebCore/css/typedom/transform/CSSRotate.idl:
* Source/WebCore/css/typedom/transform/CSSScale.idl:
* Source/WebCore/css/typedom/transform/CSSSkew.idl:
* Source/WebCore/css/typedom/transform/CSSSkewX.idl:
* Source/WebCore/css/typedom/transform/CSSSkewY.idl:
* Source/WebCore/css/typedom/transform/CSSTransformComponent.idl:
* Source/WebCore/css/typedom/transform/CSSTransformValue.idl:
* Source/WebCore/css/typedom/transform/CSSTranslate.idl:
* Source/WebCore/dom/Element+ComputedStyleMap.idl:
* Source/WebKit/WebProcess/Inspector/RemoteWebInspectorUI.cpp:
(WebKit::RemoteWebInspectorUI::RemoteWebInspectorUI):
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.cpp:
(WebKit::WebInspectorUI::WebInspectorUI):
(WebKit::WebInspectorUI::enableFrontendFeatures): Deleted.
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.h:

Canonical link: <a href="https://commits.webkit.org/276266@main">https://commits.webkit.org/276266@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6e0c8f7f6513030fbcbef161d8a00e9a6d7a47de

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44199 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23267 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46633 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46843 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40225 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46504 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27220 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20661 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/36416 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44776 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20315 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38076 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17450 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/17797 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39181 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2248 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/37521 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40390 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39469 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48452 "Built successfully") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/43756 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19178 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15752 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/43311 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20539 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/38347 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/42038 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9821 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20763 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/50838 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20165 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10289 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/css-typedom/attributeStyleMap.html:
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html><!-- webkit-test-runner [ CSSTypedOMEnabled=true ] -->
+<!DOCTYPE html><!-- webkit-test-runner -->
```
You can remove `<!-- webkit-test-runner -->` entirely here and in other tests that were only setting this pref.

nt1m commented on Source/WebKit/WebProcess/Inspector/WebInspectorUI.cpp:
```diff
@@ -54,17 +54,11 @@ Ref<WebInspectorUI> WebInspectorUI::create(WebPage& page)
 
-void WebInspectorUI::enableFrontendFeatures(WebPage& page)
-{
-    page.corePage()->settings().setCSSTypedOMEnabled(true);
```
I think we might want to keep this for CSSNestingEnabled. WI now uses CSS Nesting.

annevk replied:
But that was bad due to lack of caching, no? And it's also not set currently so it seems like that's better added separately? And leaving a no-op seems bad?

nt1m replied:
I guess we can always look up git history if we want this code back for nesting

annevk committed:
Remove CSSTypedOMEnabled preference
https://bugs.webkit.org/show_bug.cgi?id=271128

Reviewed by Tim Nguyen.

It's been stable for over a year.

* LayoutTests/css-typedom/attributeStyleMap.html:
* LayoutTests/css-typedom/css-style-value-parse.html:
* LayoutTests/css-typedom/sameobject.html:
* LayoutTests/fast/css-custom-paint/properties.html:
* LayoutTests/fast/css/css-typed-om/style-property-map-set-CSSMathSum-value.html:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/css/CSSStyleRule.idl:
* Source/WebCore/css/DOMCSSNamespace+CSSNumericFactory.idl:
* Source/WebCore/css/ElementCSSInlineStyle.idl:
* Source/WebCore/css/typedom/CSSKeywordValue.idl:
* Source/WebCore/css/typedom/CSSNumericValue.idl:
* Source/WebCore/css/typedom/CSSOMVariableReferenceValue.idl:
* Source/WebCore/css/typedom/CSSStyleImageValue.idl:
* Source/WebCore/css/typedom/CSSStyleValue.idl:
* Source/WebCore/css/typedom/CSSUnitValue.idl:
* Source/WebCore/css/typedom/CSSUnparsedValue.idl:
* Source/WebCore/css/typedom/StylePropertyMap.idl:
* Source/WebCore/css/typedom/StylePropertyMapReadOnly.idl:
* Source/WebCore/css/typedom/color/CSSColor.idl:
* Source/WebCore/css/typedom/color/CSSColorValue.idl:
* Source/WebCore/css/typedom/color/CSSHSL.idl:
* Source/WebCore/css/typedom/color/CSSHWB.idl:
* Source/WebCore/css/typedom/color/CSSLCH.idl:
* Source/WebCore/css/typedom/color/CSSLab.idl:
* Source/WebCore/css/typedom/color/CSSOKLCH.idl:
* Source/WebCore/css/typedom/color/CSSOKLab.idl:
* Source/WebCore/css/typedom/color/CSSRGB.idl:
* Source/WebCore/css/typedom/numeric/CSSMathClamp.idl:
* Source/WebCore/css/typedom/numeric/CSSMathInvert.idl:
* Source/WebCore/css/typedom/numeric/CSSMathMax.idl:
* Source/WebCore/css/typedom/numeric/CSSMathMin.idl:
* Source/WebCore/css/typedom/numeric/CSSMathNegate.idl:
* Source/WebCore/css/typedom/numeric/CSSMathProduct.idl:
* Source/WebCore/css/typedom/numeric/CSSMathSum.idl:
* Source/WebCore/css/typedom/numeric/CSSMathValue.idl:
* Source/WebCore/css/typedom/numeric/CSSNumericArray.idl:
* Source/WebCore/css/typedom/numeric/CSSNumericBaseType.idl:
* Source/WebCore/css/typedom/transform/CSSMatrixComponent.idl:
* Source/WebCore/css/typedom/transform/CSSPerspective.idl:
* Source/WebCore/css/typedom/transform/CSSRotate.idl:
* Source/WebCore/css/typedom/transform/CSSScale.idl:
* Source/WebCore/css/typedom/transform/CSSSkew.idl:
* Source/WebCore/css/typedom/transform/CSSSkewX.idl:
* Source/WebCore/css/typedom/transform/CSSSkewY.idl:
* Source/WebCore/css/typedom/transform/CSSTransformComponent.idl:
* Source/WebCore/css/typedom/transform/CSSTransformValue.idl:
* Source/WebCore/css/typedom/transform/CSSTranslate.idl:
* Source/WebCore/dom/Element+ComputedStyleMap.idl:
* Source/WebKit/WebProcess/Inspector/RemoteWebInspectorUI.cpp:
(WebKit::RemoteWebInspectorUI::RemoteWebInspectorUI):
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.cpp:
(WebKit::WebInspectorUI::WebInspectorUI):
(WebKit::WebInspectorUI::enableFrontendFeatures): Deleted.
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.h:

Canonical link: https://commits.webkit.org/276266@main
